### DISCUSSION TITLE
Internal Component Exports

### DISCUSSION THREAD

**Discussion by 381e35** [rainbow-me/rainbowkit#2147] [Feedback]
Hey I’d love if it was possible to import the components inside the modal and keep them in-page.

Our style guide is avoidant of modals as they don’t provide the best experience by obstructing their active context.

Ideally it would be possible to have the wallet list buttons and following actions built in the app instead of the app building a mini-app.

I understand modals are very common, but we are specifically looking for an option that does not force the use of modals.

Of course, the user will eventually have to interact with their wallet, which likely will be a modal/popup of its own so not having three different interaction patterns: app, rainbow modal, wallet modal — it would be very nice and give us much more control to build the experiences we are looking to :) 

Perhaps the recommended pattern is a fully custom UI? I like the base components, loading states, etc. I am solely trying to avoid the fact that it’s a modal.

**Comment by DanielSinclair** [comment]
@nftchance I think a fully Custom UI would be the way to go for you in this case. We use modals partially because we inject isolated divs into the page to prevent clashes with various style systems. I'd recommend you learn about [Wagmi Connectors](https://wagmi.sh/react/api/connectors) directly and implement your own wallet buttons and connector management, rather than relying on the RainbowKit dependency. You're welcome to borrow assets or code from RainbowKit for this.

**Reply by 381e35** [reply]
Makes sense, thanks! Will do 🤝
